### PR TITLE
PorousFlow fracture flow using a MultiApp approach

### LINKED ISSUE #17928: PorousFlow: provide MultiApp lower-dimensional fracture flow examples
## Reason

@lynnmunday , @cbolisetti , @rpodgorney , KT and I are working on simulating flow through fractured porous media using a MultiApp approach. In this work, the "fracture" is a 2D mesh living within the "matrix", which is a 3D mesh.

This will become a PorousFlow example, eventually.

We've been working in a separate repo of @cbolisetti .  I want to slowly transfer that work to MOOSE.  Initially, I'll just focus on the  markdown documentation, because when it's on another repo, it doesn't render properly and i can't put any links to MOOSE doco.

## Design

Slowly transfer the work across.  Aim to finish by sometime in July.

## Impact

Examples that users can follow when they're doing this sort of simulation.

### PR DESCRIPTION
**Pull request by WilkAndy**
Fixes #17928

### CODE REVIEW

**Comment by WilkAndy** [comment]
@lynnmunday - this is the doc and tests of the multiapp fracture flow.  You've seen basically all of it before - i've only made minor modifications.  I haven't addressed your query about order of matrix-fracture cycling and numerical instabilities.

Can you please review?

**Comment by lynnmunday** [comment]
@AhmedAlmetwally could you review this.  You can click on the "here" link


> View the site [here](https://mooseframework.inl.gov/docs/PRs/18294/site)

File: modules/porous_flow/doc/content/modules/porous_flow/multiapp_fracture_flow_introduction.md
Comment (AhmedAlmetwally) [inline modules/porous_flow/doc/content/modules/porous_flow/multiapp_fracture_flow_introduction.md:9]: This weblink is not available.

Comment (WilkAndy) [reply modules/porous_flow/doc/content/modules/porous_flow/multiapp_fracture_flow_introduction.md:9]: Yea, it won't be there till this is merged.  Do you have a better method for linking?

Comment (lynnmunday) [reply modules/porous_flow/doc/content/modules/porous_flow/multiapp_fracture_flow_introduction.md:9]: I told him to add it, I wasn't sure what was supposed to work on the moosebuild website.  Is that the same issue the non_pourousFlow documentation link has?

File: modules/porous_flow/doc/content/modules/porous_flow/multiapp_fracture_flow_diffusion.md
Comment (lynnmunday) [inline modules/porous_flow/doc/content/modules/porous_flow/multiapp_fracture_flow_diffusion.md:97]: is Delta t =0.125 correct.  Its not one of the dt's used in the figure

File: modules/porous_flow/doc/content/modules/porous_flow/multiapp_fracture_flow_diffusion.md
Comment (lynnmunday) [inline modules/porous_flow/doc/content/modules/porous_flow/multiapp_fracture_flow_diffusion.md:110]: Why does save_in work in this case?  In the previous 1D case you said not to use save_in because it is scaled by the element volume, I think.

File: modules/porous_flow/doc/content/modules/porous_flow/multiapp_fracture_flow_PorousFlow_3D.md
Comment (lynnmunday) [inline modules/porous_flow/doc/content/modules/porous_flow/multiapp_fracture_flow_PorousFlow_3D.md:77]: should it be fracture internal energy is zero?

**Review by lynnmunday — DISMISSED** [review]
I've reviewed this write-up and it is good to be merged.

**Comment by WilkAndy** [comment]
> I've reviewed this write-up and it is good to be merged.

Thanks @lynnmunday

File: modules/porous_flow/doc/content/modules/porous_flow/multiapp_fracture_flow_introduction.md
Comment (WilkAndy) [reply modules/porous_flow/doc/content/modules/porous_flow/multiapp_fracture_flow_introduction.md:9]: Yea, i want to refer to the entire directory, instead of just a particular file.  So i refer to the moose-master blob, which obviously doesn't exist until this gets merged to moose-master.

File: modules/porous_flow/doc/content/modules/porous_flow/multiapp_fracture_flow_diffusion.md
Comment (WilkAndy) [reply modules/porous_flow/doc/content/modules/porous_flow/multiapp_fracture_flow_diffusion.md:97]: I reference everything against this dt=0.125 "benchmark", so dt=0.125 doesn't appear in the figs. I've now changed the doco in a few places to emphasize this, eg `L2 error of each approach (with respect to the $\Delta t = 0.125$ benchmark) to...`

File: modules/porous_flow/doc/content/modules/porous_flow/multiapp_fracture_flow_diffusion.md
Comment (WilkAndy) [reply modules/porous_flow/doc/content/modules/porous_flow/multiapp_fracture_flow_diffusion.md:110]: True.  The `save_in` works in all cases except the previous one.  I've modified the discussion of the previous one to read:


(The inclusion of nodal volume is *exactly* what is required in the [diffusion](porous_flow/multiapp_fracture_flow_diffusion.md), [2D](porous_flow/multiapp_fracture_flow_PorousFlow_2D.md) and [3D](porous_flow/multiapp_fracture_flow_PorousFlow_3D.md) cases, since the amount of heat energy (measured in Joules) is transferred between the Apps in those cases, but in the current situation `CoupledForce` requires an energy density measured in Joules.m$^{-3}$.)

File: modules/porous_flow/doc/content/modules/porous_flow/multiapp_fracture_flow_PorousFlow_3D.md
Comment (WilkAndy) [reply modules/porous_flow/doc/content/modules/porous_flow/multiapp_fracture_flow_PorousFlow_3D.md:77]: I've expanded this to read:

PorousFlow's "matrix internal energy" quantifies the internal energy of the rock material.  Since there is essentially no rock material within the fracture, the matrix internal energy for the fracture App is zero.  ...

File: large_media
Comment (WilkAndy) [reply large_media:1]: The large_media update is deliberate

Comment (lynnmunday) [reply large_media:1]: @loganharbour  @dschwen Could you review this or recommend someone else to look it over?  It doesn't contain any code changes, it only contains a few examples with documentation for a porous flow multiapp that I've talked about at standup.  I've read through this and so has @rpodgorney

Comment (WilkAndy) [reply large_media:1]: Hoping that someone can review/merge this PR now that it passes

**Comment by rpodgorney** [comment]
I have reviewed and think it is good to go.  @loganharbour can you possibly move this along?

Great work team!

**Review by rpodgorney — DISMISSED** [review]
ready to go

**Comment by lynnmunday** [comment]
@loganharbour could you merge this if everything ok?  We have reviewed and what is written is good.  I don't know what the conflicting files are.

**Comment by WilkAndy** [comment]
I cannot invalidate "Internal App tests".  Can someone else do that when "Internal Apps" is fixed?

**Comment by rpodgorney** [comment]
@loganharbour -- can you please explain to me what exactly the issue is with this?

**Comment by loganharbour** [comment]
It should be good now; but next is broken for the time being.

**Comment by rpodgorney** [comment]
Thanks @loganharbour

**Comment by lynnmunday** [comment]
@WilkAndy I broke Bison on Thursday with a moose pr and then fixed it.  This might be failing because of that so you would need to rebase moose and push.

**Comment by lynnmunday** [comment]
@loganharbour will approve this?  Everything is passing now.  Rob and I have both reviewed it.

File: modules/porous_flow/examples/multiapp_fracture_flow/diffusion_multiapp/tests
Comment (loganharbour) [inline modules/porous_flow/examples/multiapp_fracture_flow/diffusion_multiapp/tests:30]: these are duplicate requirements that will not pass SQA if we wish to enable it on porous flow in the future

File: modules/porous_flow/examples/multiapp_fracture_flow/fracture_diffusion/tests
Comment (loganharbour) [inline modules/porous_flow/examples/multiapp_fracture_flow/fracture_diffusion/tests:6]: same

File: modules/porous_flow/examples/multiapp_fracture_flow/diffusion_multiapp/tests
Comment (WilkAndy) [reply modules/porous_flow/examples/multiapp_fracture_flow/diffusion_multiapp/tests:30]: This looks OK to me.  The first has "... heat energy is transferred between ..." and the second has "... temperature is transferred between ..."

File: modules/porous_flow/examples/multiapp_fracture_flow/fracture_diffusion/tests
Comment (WilkAndy) [reply modules/porous_flow/examples/multiapp_fracture_flow/fracture_diffusion/tests:6]: This also looks OK.  The first has "... fracture mesh conforms with the matrix mesh, without using ..." and the second has "... fracture mesh does not conform with the matrix mesh, using ..."

**Review by loganharbour — APPROVED** [review]

**Pull request merged**
